src/main/java/com/tfgm/services/TramStopService.java

```diff
@@ -0,0 +1,156 @@
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class TramStopService {
+  private final HashMap<String, TramStop> tramStopHashMap = new HashMap<>();
+  private final TramStopGraphService tramStopGraphService = new TramStopGraphService();
+
+  private final TramStopServiceUtilities utilities = new TramStopServiceUtilities();
+
+  public TramStopService(String tramDataPath) throws IOException {
+    JSONArray allTramStopData =
+        new JSONArray(
+            Files.readAllLines(Paths.get(tramDataPath)).stream()
+                .map(String::valueOf)
+                .collect(Collectors.joining()));
+
+    // System.out.println(allTramStopData);
+
+    for (int i = 0; i < allTramStopData.length(); i++) {
+      JSONObject tempStop = allTramStopData.getJSONObject(i);
+      tramStopHashMap.put(
+          tempStop.getString("tramStopName"),
+          new TramStop(
+              tempStop.getString("location"),
+              tempStop.getString("direction"),
+              tempStop.getString("line")));
+    }
+    for (int i = 0; i < allTramStopData.length(); i++) {
+      JSONObject tempStop = allTramStopData.getJSONObject(i);
+
+      TramStop currentTramStop = tramStopHashMap.get(tempStop.getString("tramStopName"));
+
+      currentTramStop.setPrevAndNextStops(
+          tempStop.getJSONArray("prevStop").toList().stream()
+              .map(tramStopHashMap::get)
+              .map(TramStopContainer::new)
+              .toArray(TramStopContainer[]::new),
+          tempStop.getJSONArray("nextStop").toList().stream()
+              .map(tramStopHashMap::get)
+              .map(TramStopContainer::new)
+              .toArray(TramStopContainer[]::new));
+    }
+
+    for (TramStop tramStop : tramStopHashMap.values()) {
+
+      System.out.println(tramStop.toString());
+      for (TramStopContainer n : tramStop.getNextStops()) {
+        TramStopContainer nextStopPrevReference =
+            Arrays.stream(n.getTramStop().getPrevStops())
+                .filter(s -> s.getTramStop().equals(tramStop))
+                .toList()
+                .get(0);
+
+        nextStopPrevReference.setTramLinkStop(n.getTramLinkStop());
+      }
+    }
+
+    // MAKE SURE YOU REMOVE APOSTROPHES AND WHITESPACE WHEN FINDING TRAMSTOP
+    for (TramStop i : tramStopHashMap.values()) {
+      System.out.println("\n\n" + i.getStopName() + i.getDirection());
+      System.out.println(i);
+
+      System.out.println("\nNextStop:");
+      for (TramStopContainer n : i.getNextStops()) {
+        System.out.println(n.getTramLinkStop());
+      }
+      System.out.println("\nPrevStop:");
+      for (TramStopContainer n : i.getPrevStops()) {
+        System.out.println(n.getTramLinkStop());
+      }
+    }
+  }
+
+  public void update() throws URISyntaxException, IOException {
+    HttpClient httpclient = HttpClients.createDefault();
+
+    URIBuilder builder = new URIBuilder("https://api.tfgm.com/odata/Metrolinks");
```

Comment:
Consider adding variables such as URIs to your application.properties for easy access